application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,196 @@
+            .setRequiredRange(3, 200)
+            .build();
+
+        Modal modal =
+                Modal.create(generateComponentId(reportedMessage, reportedMessageUrl), "Report")
+                    .addActionRow(body)
+                    .build();
+
+        event.replyModal(modal).queue();
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        long guildID = Objects.requireNonNull(event.getGuild(), "Could not retrieve the guildId.")
+            .getIdLong();
+        Optional<TextChannel> modMailAuditLog =
+                handleRequireModMailChannel(event, event.getJDA(), guildID);
+
+        if (modMailAuditLog.isEmpty()) {
+            return;
+        }
+
+        sendModMessage(event, args, modMailAuditLog.orElseThrow());
+    }
+
+    private Optional<TextChannel> handleRequireModMailChannel(ModalInteractionEvent event, JDA jda,
+            long guildID) {
+        Optional<TextChannel> modMailAuditLog = jda.getGuildById(guildID)
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+        if (modMailAuditLog.isEmpty()) {
+            event
+                .reply("Sorry, there was an issue sending your report to the moderators. We are "
+                        + "investigating.")
+                .queue();
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern {}",
+                    guildID, configModMailChannelPattern);
+        }
+        return modMailAuditLog;
+    }
+
+    private boolean handleIsOnCooldown(long userID, MessageContextInteractionEvent event) {
+        if (!isAuthorOnCooldown(userID)) {
+            return false;
+        }
+        event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+            .setEphemeral(true)
+            .queue();
+        return true;
+    }
+
+    private boolean isAuthorOnCooldown(long userId) {
+        return Optional.ofNullable(authorToLastReportInvocation.getIfPresent(userId))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    private MessageCreateAction createModMessage(String modalMessage, long userID,
+            String reportedMessage, String reportedMessageUrl, TextChannel modMailAuditLog) {
+        MessageEmbed embed = new EmbedBuilder().setTitle("Report")
+            .setDescription(("""
+                    Reported Message: **%s**
+                    [Reported Message URL](%s)
+                    Reason: **%s**""").formatted(reportedMessage, reportedMessageUrl, modalMessage))
+            .setAuthor("Author ID: %d".formatted(userID))
+            .setColor(AMBIENT_COLOR)
+            .build();
```

Comment:
i dont really like how this looks, ux wise. i think it would be cooler if it would look like the original message, but as embed , discord-wise.

so i would like to have the authors name and avatar as author of the embed, the timestamp of the original message as timestamp. and the message as content. and maybe the jump url as footer? or maybe as clickable button, like the profile-button on /modmail.

and then maybe the report reason as second embed. so all in all maybe:
so more like:
```java
// first embed, with the reported message
...
.setTitle("Report")
.setDescription(reportedMessage)
.setAuthor(authorName, ..., authorAvatar)
.setTimestamp(reportedMessageTimestamp)
.setFooter(reportedMessageUrl)
...

// second embed with the reason
...
.setTitle("Reason")
.setDescription(reportReason)
...
```